src/fobnail-attester.c

```diff
@@ -129,6 +130,149 @@ static void coap_aik_handler(struct coap_resource_t* resource, struct coap_sessi
 
 }
 
+static inline void free_null_ptr(void *ptr)
+{
+    if (ptr != NULL) {
+        free(ptr);
+        ptr = NULL;
+    }
+}
+
+UsefulBuf _encode_metadata(UsefulBuf Buffer, struct meta_data *meta)
+{
+    QCBOREncodeContext ctx;
+    QCBOREncode_Init(&ctx, Buffer);
+
+    QCBOREncode_OpenMap(&ctx);
+        QCBOREncode_AddUInt64ToMap(&ctx, "version", meta->header_version);
+        UsefulBufC mac = { meta->mac, sizeof(meta->mac) };
+        QCBOREncode_AddBytesToMap(&ctx, "mac", mac);
+
+        QCBOREncode_AddSZStringToMap(&ctx, "manufacturer",
+                                     meta->manufacturer ? meta->manufacturer : "");
+        QCBOREncode_AddSZStringToMap(&ctx, "product_name",
+                                     meta->product_name ? meta->product_name : "");
+        QCBOREncode_AddSZStringToMap(&ctx, "serial_number",
+                                     meta->serial_number ? meta->serial_number : "");
+
+    QCBOREncode_CloseMap(&ctx);
+
+    UsefulBufC EncodedCBOR;
+    QCBORError uErr;
+    uErr = QCBOREncode_Finish(&ctx, &EncodedCBOR);
+
+    if(uErr != QCBOR_SUCCESS) {
+        fprintf(stderr, "QCBOR error: %d\n", uErr);
+        return NULLUsefulBuf;
+    } else {
+        return UsefulBuf_Unconst(EncodedCBOR);
+    }
+}
+
+UsefulBuf encode_meta(struct meta_data *meta)
+{
+    UsefulBuf ret = NULLUsefulBuf;
+
+    //TODO: Error handling?
+    ret = _encode_metadata(SizeCalculateUsefulBuf, meta);
+    ret.ptr = malloc(ret.len);
+    ret = _encode_metadata(ret, meta);
+
+    return ret;
+}
+
+UsefulBuf _encode_signed_object(UsefulBuf Buffer, uint8_t *object, size_t len) {
```

Comment:
`*object` and `len` can be passed as one `UsefulBuf`

Reply:
Fixed